environ.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"os"
+	"strings"
+)
+
+const clientsKey = "XLANG_CLIENTS"
+const paramKey = "XLANG_PARAMS_"
+
+func ReadMatrixFromEnviron() Matrix {
+
+	clients := strings.Split(os.Getenv(clientsKey), ",")
+
+	params := make(Params, 0)
+	paramKeys := make(ParamKeys, 0)
```

Comment:
Don't do `make(.., 0)` for slices or maps. You're asking for zero capacity which means they'll be resized immediately.

Also, for `Params`, nil is a valid empty slice. Use that. 

(nil is not a valid empty map. It'll panic if you try to insert into it.)